beeline/src/java/org/apache/hive/beeline/Commands.java

```diff
@@ -974,86 +974,79 @@ private boolean executeInternal(String sql, boolean call) {
+          hasResults = stmnt.execute(sql);
         } else {
-          stmnt = beeLine.createStatement();
-          // In test mode we want the operation logs regardless of the settings
-          if (!beeLine.isTestMode() && beeLine.getOpts().isSilent()) {
-            hasResults = stmnt.execute(sql);
-          } else {
-            InPlaceUpdateStream.EventNotifier eventNotifier =
-                new InPlaceUpdateStream.EventNotifier();
-            logThread = new Thread(createLogRunnable(stmnt, eventNotifier));
-            logThread.setDaemon(true);
-            logThread.start();
-            if (stmnt instanceof HiveStatement) {
-              HiveStatement hiveStatement = (HiveStatement) stmnt;
-              hiveStatement.setInPlaceUpdateStream(
-                  new BeelineInPlaceUpdateStream(
-                      beeLine.getErrorStream(),
-                      eventNotifier
-                  ));
-            }
-            hasResults = stmnt.execute(sql);
-            logThread.interrupt();
-            logThread.join(DEFAULT_QUERY_PROGRESS_THREAD_TIMEOUT);
+          InPlaceUpdateStream.EventNotifier eventNotifier =
+              new InPlaceUpdateStream.EventNotifier();
+          logThread = new Thread(createLogRunnable(stmnt, eventNotifier));
+          logThread.setDaemon(true);
+          logThread.start();
+          if (stmnt instanceof HiveStatement) {
+            HiveStatement hiveStatement = (HiveStatement) stmnt;
+            hiveStatement.setInPlaceUpdateStream(
+                new BeelineInPlaceUpdateStream(
+                    beeLine.getErrorStream(),
+                    eventNotifier
+                ));
           }
+          hasResults = stmnt.execute(sql);
+          logThread.interrupt();
+          logThread.join(DEFAULT_QUERY_PROGRESS_THREAD_TIMEOUT);
         }
+      }
 
-        beeLine.showWarnings();
-
-        if (hasResults) {
-          OutputFile outputFile = beeLine.getRecordOutputFile();
-          if (beeLine.isTestMode() && outputFile != null && outputFile.isActiveConverter()) {
-            outputFile.fetchStarted();
-            if (!sql.trim().toLowerCase().startsWith("explain")) {
-              outputFile.foundQuery(true);
-            } else {
-              outputFile.foundQuery(false);
-            }
-          }
-          do {
-            ResultSet rs = stmnt.getResultSet();
-            try {
-              int count = beeLine.print(rs);
-              long end = System.currentTimeMillis();
-
-              if (showReport()) {
-                beeLine.output(beeLine.loc("rows-selected", count) + " " + beeLine.locElapsedTime(end - start),
-                    true, beeLine.getErrorStream());
-              }
-            } finally {
-              if (logThread != null) {
-                logThread.join(DEFAULT_QUERY_PROGRESS_THREAD_TIMEOUT);
-                showRemainingLogsIfAny(stmnt);
-                logThread = null;
-              }
-              rs.close();
```

Comment:
removing this line will mean that the resultset will be never closed - even when the issue doesn't happen...

isn't there a way to fix this w/o removing this close call?

Reply:
resultset::close() will close remote operation, this can be also achieved by stmnt::close() in finally block, wondering if this can be removed to make codes more elegant.

Reply:
in case it should be fixed in some other way `ResultSet` is `AutoCloseable` so it is expected to call `close` on it no matter what - I now think that this fix should either land in the jdbc driver - or should be fixed in thrift

Reply:
I've just looked around...it seems like a thrift upgrade(HIVE-21000) would need an accumulo upgrade(HIVE-22220) which would need a guava upgrade(HIVE-21569) ...

Reply:
I wondering this maybe is not a thrift problem, reading out of sequence is just a result caused by the upper request suddenly changed by some reasons. User using jdbc driver can handle the root cause at their logic as beeline does.